examples/two-tier/main.tf

```diff
@@ -49,27 +49,57 @@ resource "google_compute_instance" "www" {
     }
   }
 
-  metadata {
-    ssh-keys = "root:${file("${var.public_key_path}")}"
+  metadata = {
+    ssh-keys = "root:${file(var.public_key_path)}"
   }
 
   provisioner "file" {
-    source      = "${var.install_script_src_path}"
-    destination = "${var.install_script_dest_path}"
+    source      = var.install_script_src_path
+    destination = var.install_script_dest_path
 
     connection {
+      host = coalesce( # Simplify this to reference a specific desired IP address, if possible.
```

Comment:
This was a "TO-DO" that `terraform 0.12upgrade` generated.  My assumption is that it will stay this way as log as the access_config nat_ip is ephemeral, but please correct me if I'm wrong.